### PR TITLE
[SPARK-36109][SS][TEST] Check data after adding data to topic in KafkaSourceStressSuite

### PR DESCRIPTION
**Pull request by viirya**
### What changes were proposed in this pull request?

This patch proposes to check data after adding data to topic in `KafkaSourceStressSuite`.

### Why are the changes needed?

The test logic in `KafkaSourceStressSuite` is not stable. For example, https://github.com/apache/spark/runs/3049244904.

Once we add data to a topic and then delete the topic before checking data, the expected answer is different to retrieved data from the sink.

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

Existing tests.

### CODE REVIEW

**Comment by viirya** [comment]
cc @dongjoon-hyun @HeartSaVioR

**Comment by dongjoon-hyun** [comment]
Thank you, @viirya .

File: external/kafka-0-10-sql/src/test/scala/org/apache/spark/sql/kafka010/KafkaMicroBatchSourceSuite.scala [hunk 1]
```diff
@@ -2429,8 +2429,9 @@ class KafkaSourceStressSuite extends KafkaSourceTest {
       (d, running) => {
         Random.nextInt(5) match {
           case 0 => // Add a new topic
-            topics = topics ++ Seq(newStressTopic)
-            AddKafkaData(topics.toSet, d: _*)(message = s"Add topic $newStressTopic",
```
Comment (dongjoon-hyun) [inline external/kafka-0-10-sql/src/test/scala/org/apache/spark/sql/kafka010/KafkaMicroBatchSourceSuite.scala:2433]: So, previously, is `newStressTopic` function invoked twice?

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM.

File: sql/core/src/test/scala/org/apache/spark/sql/streaming/StreamTest.scala [hunk 2]
```diff
@@ -871,6 +871,9 @@ trait StreamTest extends QueryTest with SharedSparkSession with TimeLimits with
 
           case r if r < 0.7 => // AddData
             addRandomData()
+            // Must check data after adding data in case we delete the topic with added data
```
Comment (HeartSaVioR) [inline sql/core/src/test/scala/org/apache/spark/sql/streaming/StreamTest.scala:874]: It seems odd to explain Kafka specific one here, as the logic is a generic one. Even we need to infer which module/suite needs this with the only one hint "topic".

If we'd like to explain Kafka specific issue here, let's make clear which suite requires it.

File: external/kafka-0-10-sql/src/test/scala/org/apache/spark/sql/kafka010/KafkaMicroBatchSourceSuite.scala (hunk 1 above)
Comment (HeartSaVioR) [reply external/kafka-0-10-sql/src/test/scala/org/apache/spark/sql/kafka010/KafkaMicroBatchSourceSuite.scala:2433]: Looks like. Nice finding @viirya !

**Review by HeartSaVioR — COMMENTED** [review]
Only minor about code comment.

File: external/kafka-0-10-sql/src/test/scala/org/apache/spark/sql/kafka010/KafkaMicroBatchSourceSuite.scala (hunk 1 above)
Comment (viirya) [reply external/kafka-0-10-sql/src/test/scala/org/apache/spark/sql/kafka010/KafkaMicroBatchSourceSuite.scala:2433]: Yea, that is why we see the error message looks like:


```
AddKafkaData(topics = Set(stress4, stress6, stress2, stress1, stress5, stress3), data = Range 15 until 20, message = Add topic stress7)
```

File: sql/core/src/test/scala/org/apache/spark/sql/streaming/StreamTest.scala (hunk 2 above)
Comment (viirya) [reply sql/core/src/test/scala/org/apache/spark/sql/streaming/StreamTest.scala:874]: Updated. Thanks.

**Review by HeartSaVioR — APPROVED** [review]
+1 pending build.

**Comment by viirya** [comment]
retest this please

**Comment by dongjoon-hyun** [comment]
Thank you, @viirya and @HeartSaVioR .
GitHub Action passed. Merged to master/3.2.
Could you make a backport PR for 3.1/3.0 because this is an ancient bug, @viirya ?

**Comment by viirya** [comment]
Thanks @dongjoon-hyun and @HeartSaVioR. I will create backport PRs for 3.1/3.0.

**Pull request closed without merging**